cmake/API.cmake

```diff
@@ -138,367 +138,294 @@ function(add_fprime_subdirectory FP_SOURCE_DIR)
+#         source1.cpp
+#         source2.cpp
+#     AUTOCODER_INPUTS
+#         model.fpp
+#     DEPENDS
+#         -lm
+#     HEADERS
+#         module.h
+# )
 # ```
-# Modules requiring only autocoding may just specify *.xml files.
 #
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp)
+# > [!NOTE]  
+# > This delegates to CMake's `add_library` call. The library argument EXCLUDE_FROM_ALL is supported.
 #
-# register_fprime_module()
-# ```
+# **MODULE_NAME**: (optional) module name. Default: ${FPRIME_CURRENT_MODULE}
+# **ARGN**: sources, autocoder inputs, etc preceded by a directive (i.e. SOURCES or DEPENDS)
 #
-# ### Specific Dependencies and Linking in Modules Example ###
+####
+function(register_fprime_library)
+    fprime_add_library_build_target(${ARGN})
+    clear_historical_variables()
+    # Set up target/ targets for this module
+    fprime_attach_custom_targets("${INTERNAL_MODULE_NAME}")
+endfunction(register_fprime_library)
+
+####
+# Function `register_fprime_module`:
 #
-# Some modules need to pick a specific set of dependencies and link flags. This can be done
-# with the `MOD_DEPS` variable. This feature can be used to pick specific implementations
-# for some fprime modules that implement to a generic interface like the console logger implementation.
-# 
-# ```
-# set(SOURCE_FILE
-#     MyComponent.fpp
-#     SomeFile.cpp
-#     MyComponentImpl.cpp)
+# See `register_fprime_library`. This provides the same capability as `register_fprime_library` using the
+# backwards-compatible name. 
 #
-# set(MOD_DEPS
-#     Module1
-#     -lpthread)
+# > [!NOTE]  
+# > Variables SOURCE_FILES, MOD_DEPS, etc. are still supported but are no longer recommended.  Users are
+# > encouraged to update at their convenience.
 #
-# register_fprime_module()
-# ```
+# **MODULE_NAME**: (optional) module name. Default: ${FPRIME_CURRENT_MODULE}
+# **ARGN**: sources, autocoder inputs, etc preceded by a directive (i.e. SOURCES or DEPENDS)
 #
 ####
 function(register_fprime_module)
-    if(NOT DEFINED SOURCE_FILES)
-        message(FATAL_ERROR "'SOURCE_FILES' not defined in '${CMAKE_CURRENT_LIST_FILE}'.")
-    endif()
-    get_nearest_build_root(${CMAKE_CURRENT_LIST_DIR})
-    if (${ARGC} GREATER 0)
-        set(MODULE_NAME ${ARGV0})
-    else()
-        # Check to be sure before using
-        if (NOT DEFINED FPRIME_CURRENT_MODULE)
-            message(FATAL_ERROR "FPRIME_CURRENT_MODULE not defined. Please supply name to: register_fprime_module()")
-        endif()
-
-        set(MODULE_NAME ${FPRIME_CURRENT_MODULE})
-    endif()
-    # Explicit call to module register
-    generate_library("${MODULE_NAME}" "${SOURCE_FILES}" "${MOD_DEPS}")
-    if (TARGET "${MODULE_NAME}" AND NOT MODULE_NAME MATCHES "cmake_platform.*" )
-        add_dependencies("${MODULE_NAME}" config)
-    endif()
+    register_fprime_library(${ARGN})
 endfunction(register_fprime_module)
 
 ####
-# Function `register_fprime_executable`:
+# Function `fprime_add_library_build_target`:
```

Comment:
Question: what use case do you envision for these calls adding a module without the autocoders ?

Reply:
More of separation: add module, do something custom, then call autocoder setup.  `config` needs this flow.